File: manga_backend/app/schemas/chapter.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class ChapterResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    ChapterId: uuid.UUID
    MangaId: uuid.UUID
    Type: str | None = None
    Volume: str | None = None
    ChapterNumber: str | None = None
    Title: str | None = None
    TranslatedLang: str | None = None
    Pages: int | None = None
    PublishAt: datetime | None = None
    CreatedAt: datetime | None = None


class ChapterNav(BaseModel):
    """Returned together with chapter detail for prev/next navigation."""
    current: ChapterResponse
    prev_chapter: ChapterResponse | None = None
    next_chapter: ChapterResponse | None = None
    page_urls: list[str] = []


class ChapterGroupItem(BaseModel):
    chapter_number: str | None = None
    chapters: list[ChapterResponse] = []

File: manga_frontend/src/types/chapter.ts
import type { UUID } from "./common";

export interface Chapter {
  ChapterId: UUID;
  MangaId: UUID;
  Type?: string | null;
  Volume?: string | null;
  ChapterNumber?: string | null;
  Title?: string | null;
  TranslatedLang?: string | null;
  Pages?: number | null;
  PublishAt?: string | null;
  CreatedAt?: string | null;
}

export interface ChapterNav {
  current: Chapter;
  prev_chapter?: Chapter | null;
  next_chapter?: Chapter | null;
  page_urls: string[];
}

export interface ChapterListParams {
  lang?: string;
  sort?: "asc" | "desc";
}
